### PR TITLE
New state: favorite or unread

### LINKED ISSUE #6076: [Feature] Show both Unread and Favorite articles at same time
**Describe the solution you’d like**

When I select both "Show Unread" and "Show Favourites", I want to see all unread articles plus all favourite articles.

image

Currently, the buttons function as a filter.  When both buttons are selected, an article must be both unread and a favourite.  I would like the article filter to be logical "or" instead of "and".

Version
1.23.1

### PR DESCRIPTION
**Pull request by Alkarex**
https://github.com/FreshRSS/FreshRSS/discussions/7078#discussioncomment-11526292

New state representing *favourite OR unread* (edit: changed to `96`):

image

image

## TODO

* [ ] Add a UI button to filter on this state
* [x] Fix the existing state buttons
* [x] Add a new reading option to show *favorite OR unread* by default

### CODE REVIEW

**Comment by Alkarex** [comment]
Suggestions of UI welcome

**Comment by Alkarex** [comment]
Ah, I had completely forgotten that we already have a similar option https://github.com/FreshRSS/FreshRSS/pull/2766
image

File: .typos.toml [hunk 1]
```diff
@@ -3,6 +3,7 @@ ot = "ot"
 Ths2 = "Ths2"
 
 [default.extend-words]
+ANDS = "ANDS"
```
Comment (Frenzie) [inline .typos.toml:6]: Heh, I'd think ANS to be a more likely typo. :-)

**Comment by Alkarex** [comment]
Tests welcome @Frenzie @math-GH

**Comment by Frenzie** [comment]
State 96 works as described afaict.

**Comment by Alkarex** [comment]
> State 96 works as described afaict.

@Frenzie And what do you think about it, and of having it by default (i.e. also for state=0, which before was only showing unreads)?

**Comment by Frenzie** [comment]
That doesn't make much sense to me. Unread is unread, indicating that which you haven't yet dealt with. I consider that a very sensible default. Favorite is for items you want to be able to find again more easily than others. If you still want to actively deal with them, mark them as unread / don't mark them as read. Favorite is unrelated.

**Comment by ZicPL** [comment]
@Alkarex @Frenzie Is it possible to make "keep" read and show? It won't occupy the "Favourites" category and it won't be the same as read and not show (hide).

**Comment by Alkarex** [comment]
> Is it possible to make "keep" read and show? It won't occupy the "Favourites" category and it won't be the same as read and not show (hide).

@ZicPL I am not sure to understand your question. Have you observed in this PR a problematic change?

**Comment by ZicPL** [comment]
> (..) Have you observed in this PR a problematic change?

@Alkarex No, that isn't problem with this PR.
I suggested creating new **"keep"** category similar to **"Favourites"**. This PR concerns filter: "unread" and "Favourites". **"Keep"** as opposed to **"Favourites"** would mean: article has been **"read"** and **"remember"**. "Favourites" means for me something **important** - something **special**. **"Keep"** could have a padlock symbol and doesn't have to be on list (panel on left side "Normal View"). I also wrote about it [here](https://github.com/FreshRSS/FreshRSS/discussions/7078#discussioncomment-11534403).

I'm not sure if this would require a new (separate) Pull Request.

**Comment by Alkarex** [comment]
This is unrelated to this PR

**Comment by math-GH** [comment]
What is the difference between state `36` and `6`?

**Comment by Frenzie** [comment]
6 is favorites that are unread (favorite AND unread)

96 is all unread as well as favorites that are read (unread OR favorite)

**Comment by ZicPL** [comment]
@Alkarex @Frenzie
Is it possible in another PR?
Does this require another discussion?
Do You think it's worth making "keep" feature (in another PR)?

ps. I'm not pushing for immediate implementation. I'd like to know Your opinion: Do You positively evaluate this feature ("keep") and Is this feature "keep" a better solution than the feature from this PR.

**Comment by Frenzie** [comment]
I don't know if that's a new discussion or simply part of #7078 but it doesn't belong here. ;-)

**Comment by Alkarex** [comment]
@ZicPL This discussion is out of topic here. Feel free to open another discussion if there is isn't already one. No, at a first glance, I do not think this is a feature to prioritize as there are already favourites and labels for keeping articles, and I would rather avoid adding some unneeded technical and usability complexity

**Comment by ZicPL** [comment]
Thanks for Your feedback. Feature from current PR implements goals from our last discussion. I'll be testing this and if I notice any issues I'll report here.

**Comment by Alkarex** [comment]
New option implemented

image

**Comment by Alkarex** [comment]
Side note: we should make the sate of the state buttons more visually distinct
image

Especially on mobile with bad lighting, I often have difficulties distinguishing between an enabled and disabled state

**Comment by Alkarex** [comment]
Besides a bit more testing, I think we could stop here for this PR when it comes to FreshRSS 1.25.0 as I believe it covers the initial demand. Adding a button to manually change state to *favourite or unread* can wait for a future improvement.

**Comment by Frenzie** [comment]
The color should probably be more muted while disabled. Depending on how new we want our CSS that's actually really easy now.

File: app/i18n/it/conf.php [hunk 2]
```diff
@@ -257,12 +257,13 @@
 		'show' => array(
 			'_' => 'Articoli da visualizzare',
 			'active_category' => 'Categoria attiva',
-			'adaptive' => 'Adatta visualizzazione',
+			'adaptive' => 'Show unreads if any, all articles otherwise',	// TODO
```
Comment (UserRoot-Luca) [inline app/i18n/it/conf.php]:
```suggestion
			'adaptive' => 'Mostra i non letti se ci sono o tutti gli articoli altrimenti',
```

File: app/i18n/it/conf.php [hunk 3]
```diff
@@ -257,12 +257,13 @@
...
 			'all_articles' => 'Mostra tutti gli articoli',
 			'all_categories' => 'Tutte le categorie',
 			'no_category' => 'Nessuna categoria',
 			'remember_categories' => 'Ricorda le categorie aperte',
 			'unread' => 'Mostra solo non letti',
+			'unread_or_favorite' => 'Show unreads and favourites',	// TODO
```
Comment (UserRoot-Luca) [inline app/i18n/it/conf.php]:
```suggestion
			'unread_or_favorite' => 'Mostra i non letti e i preferiti',	
```

File: app/i18n/pl/conf.php [hunk 4]
```diff
@@ -257,12 +257,13 @@
 		'show' => array(
 			'_' => 'Wiadomości do wyświetlenia',
 			'active_category' => 'Aktualna',
-			'adaptive' => 'Dopasuj do sytuacji',
+			'adaptive' => 'Show unreads if any, all articles otherwise',	// TODO
```
Comment (ZicPL) [inline app/i18n/pl/conf.php]:
```suggestion
			'adaptive' => 'Pokaż istniejące nieprzeczytane artykuły, w przeciwnym razie wyświetl wszystkie',	// TODO
```

File: app/i18n/pl/conf.php [hunk 5]
```diff
@@ -257,12 +257,13 @@
...
 			'all_articles' => 'Wszystkie wiadomości',
 			'all_categories' => 'Wszystkie',
 			'no_category' => 'Żadna',
 			'remember_categories' => 'Pamiętaj otwarte kategorie',
 			'unread' => 'Tylko nieprzeczytane',
+			'unread_or_favorite' => 'Show unreads and favourites',	// TODO
```
Comment (ZicPL) [inline app/i18n/pl/conf.php]:
```suggestion
			'unread_or_favorite' => 'Pokaż nieprzeczytane i ulubione',	// TODO
```

**Pull request merged**